frontend/src/core/user/components/UserNotificationsMenu.js

```diff
@@ -22,17 +23,27 @@ type State = {|
 |};
 
 type UserNotificationsMenuProps = {
+    has_unread: boolean,
+    logUxAction: (string, ?string, ?any) => void,
     notifications: Array<Notification>,
     onDiscard: (e: SyntheticEvent<any>) => void,
 };
 
 export function UserNotificationsMenu({
+    has_unread,
+    logUxAction,
     notifications,
     onDiscard,
 }: UserNotificationsMenuProps): React.Element<'div'> {
     const ref = React.useRef(null);
     useOnDiscard(ref, onDiscard);
 
+    function handleClickSeeAll() {
+        logUxAction('See all Notifications link clicked', 'Notifications 1.0', {
+            unread: has_unread,
+        });
```

Comment:
This needs tweaking (also on the old frontend): right now, the request could get interrupted by redirecting to `/notifications`, in which case the action will not be logged.

Reply:
How about just adding a flag to the link on the buttons, and do the ux action when loading the all-notifications page? That way, you don't need the UX state (I guess it'd need to be state), and we'd have one data set for people using UI controls vs just loading the page from history, too.

I'd suggest to unset the flag on load of the notifications page, though, so that reloads get flagged as just loads, and not UI control actions.